test/test_testing.py

```diff
@@ -101,14 +101,14 @@ def test_add_relation(self):
         assert backend.relation_get(rel_id, 'test-app', is_app=True) == {}
         assert backend.relation_get(rel_id, 'test-app/0', is_app=False) == {}
 
-    def test_add_relation_with_app_data(self):
+    def test_add_relation_with_app_data(self, request: pytest.FixtureRequest):
         harness = ops.testing.Harness(ops.CharmBase, meta='''
             name: test-app
             requires:
                 db:
                     interface: pgsql
             ''')
-        self.addCleanup(harness.cleanup)
+        request.addfinalizer(harness.cleanup)
```

Comment:
I wonder if something like a `harness_factory` or `self_cleaning_harness` fixture would be more idiomatic to use. What I mean is some way to avoid the repetitive `.addfinalizer` calls.

Reply:
If `harness` is a fixture, this wouldn't be an issue because we can add cleanup after `yield`. However, since the meta is different for different test cases, we did not want to make `harness` as a fixture here, because passing parameters (meta) to a fixture can be less-readable.

Resolving this comment with the above explanation.